packages/experiments/src/components/Button/Button.state.ts

```diff
@@ -0,0 +1,23 @@
+import * as React from 'react';
+import { BaseState } from '../../utilities/BaseState';
+import { IButtonProps, IButtonViewProps } from './Button.types';
+
+export type IButtonState = Pick<IButtonViewProps, 'buttonRef'>;
+
+export class ButtonState extends BaseState<IButtonProps, IButtonViewProps, IButtonState> {
+  private _buttonRef = React.createRef<HTMLButtonElement>();
+
+  constructor(props: ButtonState['props']) {
+    super(props);
+
+    this.state = {
+      buttonRef: this._buttonRef
```

Comment:
It sucks to have to add state just for this, particularly since it doesn't affect styling. I wonder if we should just implement this as a hook in view for now, knowing that it doesn't affect styling. I'll add this to #7014 as a point to consider.

Reply:
https://reactjs.org/docs/hooks-reference.html#useimperativehandle

Reply:
I *think* something like this makes sense:

```tsx
createComponent({
  state: (userProps) => viewProps,
  ...
});
```

Then we can create reusable hooks for everything and use the state to collect those things to transform user props into the props the view needs.

Reply:
If we implement this via hooks it would be in a separate PR given that we'd have to target the fabric 7 branch instead of the master branch.

Reply:
We'll have to start working there sooner rather than later, particularly with my slots API changes which will all be on `fabric-7`. Next item up is the hook / state resolution anyways, which will also be on `fabric-7`.